pennylane_lightning/core/src/observables/Observables.hpp

```diff
@@ -220,10 +218,8 @@ class HermitianObsBase : public Observable<StateVectorT> {
     MatrixT matrix_;
     std::vector<std::size_t> wires_;
 
-#ifdef PL_USE_LAPACK
-    std::vector<PrecisionT> eigenVals_;
-    MatrixT unitary_;
-#endif
+    mutable std::vector<PrecisionT> eigenVals_;
+    mutable MatrixT unitary_;
```

Comment:
What's the advantage of `mutable` here? :thinking:

Reply:
We have to set it `mutable` as `obs` objects are always set as `const` and the `unitary_` is calculated when calling  `applyInShots` at runtime, which means it should be mutable.